File: pset6/sentimental-readability/readability.py
import cs50


def main():
    text = cs50.get_string("Text: ")

    L = 100 * (letters(text) / words(text))
    S = 100 * (sentences(text) / words(text))
    index = round(0.0588 * (L) - 0.296 * (S) - 15.8)

    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade ", index)


def letters(text):
    count = 0
    for i in range(len(text)):
        if text[i].isalpha():
            count += 1
    return count


def words(text):
    count = 0
    for i in range(len(text)):
        if text[i].isspace():
            count += 1
    return count + 1


def sentences(text):
    count = 0
    for i in range(len(text)):
        if text[i] in ['?', '.', '!']:
            count += 1
    return count


if __name__ == "__main__":
    main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int letters(string text);
int words(string text);
int sentences(string text);

int main(void)
{
    string text = get_string("Text: ");

    int Letters = letters(text);
    int Words = words(text);
    int Sentences = sentences(text);

    float L = (float) Letters / Words * 100;
    float S = (float) Sentences / Words * 100;

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

int letters(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}

int words(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if (isspace(text[i]))
        {
            count++;
        }
    }
    return count + 1;
}

int sentences(string text)
{
    int count = 0;
    for (int i = 0; i < strlen(text); i++)
    {
        if ((text[i]) == '.' || (text[i]) == '!' || (text[i]) == '?')
        {
            count++;
        }
    }
    return count;
}